package/kedro_viz/launchers/cli/run.py

```diff
@@ -145,6 +146,9 @@ def run(
             "https://github.com/kedro-org/kedro-viz/releases.",
             "yellow",
         )
+
+    port = _find_available_port(host, port)
```

Comment:
This looks good but should we also investigate and fix shutting down of previous viz process efficiently once the user exits ?

Reply:
Good one, but that maybe out of scope for this particular ticket. and we will also need to think what would be the ideal user journey. Should we prompt them to kill the process, should we ask them for consent to kill the process etc. 

@ravi-kumar-pilla , could u create another issue
 for this?

Reply:
> Good one, but that maybe out of scope for this particular ticket. and we will also need to think what would be the ideal user journey. Should we prompt them to kill the process, should we ask them for consent to kill the process etc.

Not about the user flow, I was saying may be we need to know the root cause of why a kedro viz process was hanging and we need to shutdown kedro viz gracefully (i.e., when someone did ctrl+c, make sure the process is shut down, this also includes kedro viz process started via jupyter notebook). Yes I can create a separate ticket but I thought it is related to #2052

Reply:
n the #2052  case described by Juanlu, he was probably trying to compare kedro-viz and kedro-viz --lite , so he wasn't terminating the previous process.

Reply:
In Juanlu's case he tried to start two viz processes and didn't exist the first instance, the second instance would fail to start on same port but still opened the browser and redirected to the first instance.

Reply:
got it, thank you @rashidakanchwala @SajidAlamQB

Reply:
> Should we prompt them to kill the process, should we ask them for consent to kill the process etc.

Notice that the process keeping the port busy can be anything. We shouldn't meddle with that, either say "sorry, that port isn't available, use another one" or choose another one ourselves (either at random or incrementing by +1 as I think this PR does)

Reply:
I completely agree that we must not interfere with the user's machine. So, ideally it would be informing them, and letting them close it themselves which this PR does.